netbox/dcim/models/device_components.py

```diff
@@ -30,6 +30,35 @@
     'RearPort',
 )
 
+QUERY_POWER_DRAW = """
```

Comment:
We need to avoid raw SQL wherever possible as it is very difficult to maintain long term. (It is completely opaque to schema migrations and exempt from IDE introspection.) If we can't find a way to do it efficiently using Django's ORM, we have to ask whether the effort is worth it. I'll defer to the other maintainers, but personally I'm opposed to the introduction of any new raw SQL, anywhere in the project. (We have some bits we're still trying to get rid of.)

Reply:
I agree, though I simply couldn't find a native way of performing recursion through the ORM. You'll notice the affects of this when you're on the racks list and have lots of power feeds assigned to them as each of those power feeds' attached power port is called with `get_power_draw`.

We can do this through the ORM, but the best that I got is _O(n)_ with _n_ being the depth (i.e. hops from the origin).

Reply:
After some discussion with the other maintainers, we've agreed to reject the introduction of any new raw SQL queries if they can be avoided, for the reasons I mentioned above. It's simply not an acceptable development burden given the current resources afforded to the project. If this feature is to be implemented, it will need to be accomplished using the Django ORM.